client/post-editor/editor-featured-image/index.jsx

```diff
@@ -115,6 +127,7 @@ export default React.createClass( {
 				borderless
 				compact
 				className={ classes }>
+				{ site && featuredImageId && <QueryMedia siteId={ site.ID } mediaId={ featuredImageId } /> }
```

Comment:
Looks like we should also update checks in QueryMedia to do a no-op if props aren't available like siteId